core/iwasm/interpreter/wasm_interp_classic.c

```diff
@@ -891,7 +891,7 @@ wasm_interp_call_func_native(WASMModuleInstance *module_inst,
     if (!func_import->call_conv_wasm_c_api) {
         native_func_pointer = module_inst->import_func_ptrs[cur_func_index];
     }
-    else {
+    else if (module_inst->e->c_api_func_imports) {
         c_api_func_import = module_inst->e->c_api_func_imports + cur_func_index;
         native_func_pointer = c_api_func_import->func_ptr_linked;
     }
```

Comment:
Guess a more `else` branch about the condition of `func_import->call_conv_wasm_c_api == true && module_inst->e->c_api_func_imports == NULL` would  be better

Reply:
If another else branch is added, I think what we can do is to set native_func_pointer to NULL inside it:
```C
    if (!func_import->call_conv_wasm_c_api) {
        native_func_pointer = module_inst->import_func_ptrs[cur_func_index];
    }
    else if (module_inst->e->c_api_func_imports) {
        c_api_func_import = module_inst->e->c_api_func_imports + cur_func_index;
        native_func_pointer = c_api_func_import->func_ptr_linked;
    }
    else {
       native_func_pointer = NULL;
    }
```
But the else branch can do nothing since native_func_pointer is set to NULL at the function beginning.

Reply:
IIUC, `wasm_instance_new` shall set `call_conv_wasm_c_api` to `true` and bind `c_api_func_imports` if execute successful. So, a warning or error should be raised.

Reply:
Do you mean that the `module_inst->e->c_api_func_imports` must not be NULL when wasm_instance_new is called and the import func count is not 0? I guess here the issue reported by sanitizers in aot_runtime.c is that the developer uses the APIs in wasm_export.h to load and instantiate the wasm module, and wasm-c-api wasm_instance_new isn't called, so `module_inst->e->c_api_func_imports` in aot_runtime.c is NULL, and `c_api_func_imports + func_idx` triggers sanitizer error.

Seems that we only need to add check in aot_runtime.c:
```C
    CApiFuncImport *c_api_func_import =
        module_inst_extra->c_api_func_imports
            ? module_inst_extra->c_api_func_imports + func_idx
            : NULL;
```
And in other places when call_conv_wasm_c_api is detected as true, no need to change the code, or just add an assertion, e.g.:
```C
    if (import_func->call_conv_wasm_c_api) {
        bh_assert(c_api_func_import);
        func_ptr = c_api_func_import->func_ptr_linked;
    }
```

Reply:
Got it. In that case (`wasm_instance_new` isn't called), it does make sense.

LGTM now.